language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ConvertToConfigurableCodeAction.java

```diff
@@ -0,0 +1,214 @@
+            return false;
+        }
+
+        if (!variableSymbol.typeDescriptor().subtypeOf(semanticModel.types().READONLY)) {
+            switch (matchedNode.kind()) {
+                case LIST_CONSTRUCTOR -> {
+                    return !isListVarUpdated(references, rootNode);
+                }
+                case MAPPING_CONSTRUCTOR -> {
+                    return !isMappingVarUpdated(references, rootNode);
+                }
+                case TABLE_CONSTRUCTOR -> {
+                    return !isTableVarUpdated(references, rootNode);
+                }
+            }
+        }
+
+        return true;
+    }
+
+    @Override
+    public List<CodeAction> getCodeActions(CodeActionContext context, RangeBasedPositionDetails posDetails) {
+        ModuleVariableDeclarationNode modVarDeclarationNode =
+                (ModuleVariableDeclarationNode) posDetails.matchedCodeActionNode().parent();
+
+        TextEdit textEdit = modVarDeclarationNode.visibilityQualifier()
+                .map(token ->
+                        new TextEdit(PositionUtil.toRange(token.lineRange().endLine()), " " + CONFIGURABLE))
+                .orElseGet(() ->
+                        new TextEdit(PositionUtil.toRange(modVarDeclarationNode.lineRange().startLine()),
+                                CONFIGURABLE + " "));
+
+        CodeAction codeAction =  CodeActionUtil.createCodeAction(CommandConstants.CONVERT_TO_CONFIGURABLE,
+                List.of(textEdit), context.fileUri(), CodeActionKind.Empty);
+        return List.of(codeAction);
+    }
+
+    private static boolean hasIsolatedOrFinalOrConfigurableQualifier(
+            ModuleVariableDeclarationNode modVarDeclarationNode) {
+        return modVarDeclarationNode.qualifiers()
+                .stream().anyMatch(q -> q.text().equals(Qualifier.ISOLATED.getValue())
+                        || q.text().equals(Qualifier.FINAL.getValue())
+                        || q.text().equals(Qualifier.CONFIGURABLE.getValue()));
+    }
+
+    private static boolean isVariableAssigned(List<Location> references, NonTerminalNode rootNode) {
+        return references.stream()
+                .map(location -> rootNode.findNode(location.textRange()))
+                .anyMatch(node -> node.parent().kind() == SyntaxKind.ASSIGNMENT_STATEMENT);
+    }
+
+    private static boolean isListVarUpdated(List<Location> references, NonTerminalNode rootNode) {
+        for (Location location : references) {
+            NonTerminalNode node = rootNode.findNode(location.textRange());
+            NonTerminalNode parentNode = node.parent();
+            SyntaxKind parentKind = parentNode.kind();
+            if (parentKind == SyntaxKind.METHOD_CALL) {
+                MethodCallExpressionNode methodCallExpressionNode = (MethodCallExpressionNode) parentNode;
+                if (LIST_UPDATING_METHODS.contains(methodCallExpressionNode.methodName().toSourceCode())) {
+                    return true;
+                }
```

Comment:
Can't we merge these two logics? we don't necessarily need to have a separate variable for `methodCallExpressionNode`.